[DOCS] TSVB updates for 7.12 (#93317)
elastic/kibana: docs/user/dashboard/tsvb.asciidoc

[[TSVB]]
=== TSVB

*TSVB* enables you to visualize the data from multiple data series, supports <<aggregation-reference, 

> most {es} metric aggregations>>, multiple visualization types, custom functions, and some math. To use *TSVB*, you must have a date field. 

Reviewer: you must have a date field > your data must have a date field?

Is it date field or time field?

Author: This is content that Wylie wrote, so I was hoping he would have commented. Since he is out this week, I'm going to leave it, but I'll make a note to follow up for the 7.13 release.

[role="screenshot"]
image::visualize/images/tsvb-screenshot.png[TSVB overview]

[float]
[[tsvb-required-choices]]
==== Open and set up TSVB

Open *TSVB*, then make sure the required settings are configured.

. On the dashboard, click *Create panel*.

. On the *New visualization* window, click *TSVB*.

. In *TSVB*, click *Panel options*, then make sure the following settings are configured:

* *Index pattern*
* *Time field*
* *Interval*

> . Select a *Drop last bucket* option. Typically, the last bucket contains partial data.

Reviewer 2: Small clarification here.

```suggestion
. Select a *Drop last bucket* option. It is dropped by default because the time filter intersects the time range of the last bucket, but can be enabled to see the partial data.
```

. In the *Panel filter* field, specify any <<kuery-query, KQL filters>> to select specific documents.

[float]
[[configure-the-data-series]]
==== Configure the series

Each *TSVB* visualization shares the same options to create a *Series*. Each series can be thought of as a separate {es} aggregation, which prevents

> them from being compared with math. For each series, the *Options* control the styling and {es} options, and are inherited from *Panel options*.

Reviewer: what does "them" refer to?

When you have separate options for each series, you can compare different {es} indices, and view two time ranges from the same index. 

To configure the value of each series, select the function, then configure the function inputs. Only the last function is displayed.

. From the *Aggregation* dropdown, select the function for the series. 

. To display each group separately, select one of the following options from the *Group by* dropdown:

* *Filters* &mdash; Groups the data into the specified filters. To differentiate the groups, assign a color to each filter.

> * *Terms* &mdash; Displays the top values in the series. To configure for the *Time Series* visualization, click *Options*, then select an option from the *Split color theme* dropdown.

Reviewer 2: ```suggestion
* *Terms* &mdash; Displays the top values of the field. The color is only configurable in the *Time Series* chart. To configure, click *Options*, then select an option from the *Split color theme* dropdown.
```

[float]
[[tsvb-time-series]]
===== Time Series

By default, the y-axis displays the full range of data, including zero. To automatically scale the y-axis from
the minimum to maximum values of the data, click *Data > Options > Fill*, then enter `0` in the *Fill* field.
You can add annotations to the x-axis based on timestamped documents in a separate {es} index.

[float]
[[all-chart-types-except-time-series]]
===== All chart types except Time Series

The *Data timerange mode* dropdown in *Panel options* controls the timespan that *TSVB* uses to match documents.
*Last value* is unable to match all documents, only the specific interval. *Entire timerange* matches all documents specified in the time filter.

[float]
[[metric-topn-gauge]]
===== Metric, Top N, and Gauge

*Color rules* in *Panel options* contains conditional coloring based on the values. 

[float]
[[topn-table]]
===== Top N and Table

When you click a series, *TSVB* applies a filter based on the series name. 
To change this behavior, click *Panel options*, then specify a URL in the *Item URL* field, which opens a URL instead of applying a filter on click. 

[float]
[[tsvb-markdown]]
===== Markdown

> Supports Markdown with Handlebar (mustache) syntax to insert dynamic data, and supports custom CSS using the LESS syntax.

Reviewer: maybe this should be a full sentence to be parallel with the others

==== TSVB function reference

*TSVB* provides you with shortcuts for some frequently-used functions.

*Filter Ratio*::
  Returns a percent value by calculating a metric on two sets of documents. 
  For example, calculate the error rate as a percentage of the overall events over time.

*Counter Rate*::
  Used when dealing with monotonically increasing counters. Shortcut for *Max*, *Derivative*, and *Positive Only*.

*Positive Only*::
  Removes any negative values from the results, which can be used as a post-processing step
  after a derivative.

*Series Agg*::
  Applies a function to all of the *Group by* series to reduce the values to a single number.
  This function must always be the last metric in the series.
  For example, if the *Time Series* visualization shows 10 series, the sum *Series Agg* calculates
  the sum of all 10 bars and output a single Y value per X value. This is often confused
  with the overall sum function, which outputs a single Y value per unique series.

*Math*::
  The math context is able to do simple and advanced calculations per series.
  This function must always be the last metric in the series.

[float]
[[tsvb-faq]]
==== Frequently asked questions

> For answers to frequently asked *TSVB* question, review the following. 

Reviewer 2: Is this sentence necessary?

Author: Is the content necessary? No. But Asciidoc requires some sort of wording to separate titles.

[float]
===== Why is my TSVB visualization missing data?

It depends, but most often there are two causes:

* For *Time series* visualizations with a derivative function, the time interval can be too small. Derivatives require sequential values. 

* For all other *TSVB* visualizations, the cause is probably the *Data timerange mode*, which is controlled by *Panel options > Data timerange mode > Entire time range*.
By default, *TSVB* displays the last whole bucket. For example, if the time filter is set to *Last 24 hours*, and the
current time is 9:41, *TSVB* displays only the last 10 minutes &mdash; from 9:30 to 9:40.

[float]
===== How do I calculate the difference between two data series?

Performing math across data series is unsupported in *TSVB*. To calculate the difference between two data series, use <<create-panels-with-timelion, Timelion>> or <<vega, Vega>>.

[float]
===== How do I compare the current versus previous month?

> While math is unsupported in *TSVB*, series with tie offsets is supported.

Reviewer 2: Clarifying this because TSVB does support math, just not this kind of math.

```suggestion
*TSVB* can display two series with time offsets, but it can't perform math across series. To add a time offset:
```
